### PR TITLE
DOC: improve docstring of function where

### PR DESCRIPTION
**Pull request by JennaVergeynst**
Clarified parameters `cond` and `other`.
Added example with scalar `other`.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5822,20 +5822,24 @@ def _where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
```
Comment (jreback) [inline pandas/core/generic.py]: can you update same for ``.mask`` as well.

Comment (jorisvandenbossche) [reply pandas/core/generic.py]: mask and where now use the same docstring (with only parts of it substituted with the correct value). But keep doing this and providing a better explanation will become rather complex I think. Are you OK with just duplicating the docstring and have one for where and mask separately ?

Comment (jreback) [reply pandas/core/generic.py]: oh that's right. I don't think duplicating is a good idea. maybe just put an example with mask here as well.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jreback** [comment]
thanks @JennaVergeynst

**Pull request merged**